### PR TITLE
created header component and added to App.js

### PR DESCRIPTION
**Pull request by galadoe**
I'm having trouble importing the paths. not too sure if the NavLink is going to the correct place

### CODE REVIEW

**Review by WalkerTexas-coder — COMMENTED** [review]
Just to clarify you are looking to delete everything in App.js other than header?

**Comment by galadoe** [comment]
yes

File: app/javascript/components/App.js [hunk 1]
```diff
@@ -23,37 +24,7 @@ class App extends React.Component {
...
-              <NavLink to="/learn">Learn More</NavLink>
-            </NavItem>
-          </Nav>
-          <Switch>
-            <Route exact path="/" component={ Home } />
-            <Route path="/about" component={ AboutUs } />
-            <Route path="/learn" component={ LearnMore } />
-          </Switch>
-        </Router>
+        <Header />
```
Comment (WalkerTexas-coder) [inline app/javascript/components/App.js]: okay heads up you aren't passing any props down to header here. Have you checked that this set up works as is?

Comment (galadoe) [reply app/javascript/components/App.js]: it works

**Review by sjproctor — CHANGES_REQUESTED** [review]
Take another pass at this. This looks like code from the exploratory app and not from the current project.

File: app/javascript/components/App.js [hunk 2]
```diff
@@ -1,15 +1,36 @@
 import React from "react"
 import PropTypes from "prop-types"
-import { BrowserRouter as  Router, Route, Switch, NavLink } from "react-router-dom"
-
-import { Nav, NavItem } from "reactstrap"
+// import { BrowserRouter as  Router, Route, Switch, NavLink } from "react-router-dom"
+//
+// import { Nav, NavItem } from "reactstrap"
 // import "application.scss"
```
Comment (sjproctor) [inline app/javascript/components/App.js:6]: You will not need to import the scss file.

File: app/javascript/components/App.js [hunk 3]
```diff
@@ -1,15 +1,36 @@
 import React from "react"
 import PropTypes from "prop-types"
-import { BrowserRouter as  Router, Route, Switch, NavLink } from "react-router-dom"
-
-import { Nav, NavItem } from "reactstrap"
+// import { BrowserRouter as  Router, Route, Switch, NavLink } from "react-router-dom"
```
Comment (sjproctor) [inline app/javascript/components/App.js]: You will need your imports for react-router-dom

File: app/javascript/components/App.js [hunk 4]
```diff
@@ -1,15 +1,36 @@
...
+    }
+  }
+
+  //render index with updated apartment array
+  componentDidMount(){
+    this.apartmentIndex()
+  }
+
+  apartmentIndex = () => {
+    fetch("http://localhost:3000/apartments")
```
Comment (sjproctor) [inline app/javascript/components/App.js]: Your fetch can happen to just `"/apartments"`

File: app/javascript/components/components/Header.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+              <a href={sign_out_route}>Sign Out</a>
+            </div>
+        }
+        {
+          !logged_in &&
+            <div>
+              <a href={sign_in_route}>Sign In</a>
+            </div>
+        }
+        <Router>
```
Comment (sjproctor) [inline app/javascript/components/components/Header.js:21]: You should only have one router per application and your router is in App.js. This is also code that you should not have in this application as this is not from the apartment app project.

File: app/views/home/index.html.erb [hunk 6]
```diff
@@ -3,6 +3,7 @@ logged_in: user_signed_in?,
 current_user: current_user,
 new_user_route: new_user_registration_path,
 sign_in_route: new_user_session_path,
-sign_out_route: destroy_user_session_path
+sign_out_route: destroy_user_session_path,
+apartments_route: apartment_path
```
Comment (sjproctor) [inline app/views/home/index.html.erb]: What is line 7 doing?

File: spec/requests/apartments_spec.rb [hunk 7]
```diff
@@ -3,5 +3,33 @@
 RSpec.describe "Apartments", type: :request do
   describe "GET /index" do
     pending "add some examples (or delete) #{__FILE__}"
```
Comment (sjproctor) [inline spec/requests/apartments_spec.rb]: Line 5 should be removed

**Review by sjproctor — CHANGES_REQUESTED** [review]
Getting close. Still a few things to do.

File: app/javascript/components/App.js (hunk 3 above)
Comment (galadoe) [reply app/javascript/components/App.js]: Imported for react-router-dom

File: app/javascript/components/components/Header.js (hunk 5 above)
Comment (galadoe) [reply app/javascript/components/components/Header.js:21]: I removed it

File: spec/requests/apartments_spec.rb (hunk 7 above)
Comment (galadoe) [reply spec/requests/apartments_spec.rb]: removed it

File: app/javascript/components/App.js (hunk 2 above)
Comment (galadoe) [reply app/javascript/components/App.js:6]: okay

File: app/javascript/components/App.js (hunk 4 above)
Comment (galadoe) [reply app/javascript/components/App.js]: okay

**Comment by galadoe** [comment]
When I run the code and look at the console. It says that ./pages/Home modules cannot be found. I know that I'm importing it correctly.

File: app/javascript/components/App.js [hunk 8]
```diff
@@ -2,14 +2,35 @@ import React from "react"
...
+    }
+  }
+
+  //render index with updated apartment array
+  componentDidMount(){
+    this.apartmentIndex()
+  }
+
+  apartmentIndex = () => {
+    fetch("http://localhost:3000/apartment")
```
Comment (sjproctor) [inline app/javascript/components/App.js]: Your fetch should be to just `"/apartments"`

File: app/javascript/components/components/Header.js [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+      new_user_route,
+      sign_in_route,
+      sign_out_route
+    } = this.props
+
+    return(
+      <>
+        <Router>
+          <br/>
+          <Nav>
```
Comment (sjproctor) [inline app/javascript/components/components/Header.js:23]: Just double checking that this code is removed. No router in Header.

**Review by sjproctor — APPROVED** [review]
Looks good!

**Pull request merged**